intake: Clean up and fix names

auscompgeek opened the pull request:
The terrible naming was _really_ getting on my nerves.

Looking at IntakeAutomation, I don't think the `on_enable` part is the right place to put that... but I'm not sure what a nice solution for that would look like.

auscompgeek committed:
intake: Clean up and fix names

ArthurAllshire commented on automations/intake.py:
```diff
@@ -1,80 +1,72 @@
-            self.intake.intake_clamp(False)
+        """Start the intake motors and wait for the infrared sensor to see the cube."""
+        if self.intake.is_seeing_cube() and state_tm > 0.5:
+            self.intake.clamp(False)
```
Do we not want to also pull in the kicker in this state?

auscompgeek replied:
That sounds like a good idea.

ArthurAllshire commented on components/intake.py:
```diff
@@ -31,61 +33,40 @@ def on_disable(self):
-            return True
-        else:
-            return False
+    def is_seeing_cube(self) -> bool:
```
I think that this name is still fairly non-descriptive - its not checking whether we are seeing the cube, it is checking that the cube is in the containment mech.

auscompgeek replied:
`is_cube_contained` sounds like a much better name. I really wasn't sure what this was actually being used for from the name nor the docstring here. Will fix.

ArthurAllshire commented on components/intake.py:
```diff
@@ -31,61 +33,40 @@ def on_disable(self):
-            return True
-        else:
-            return False
+    def is_contacting_cube(self) -> bool:
```
What is contacting the cube? This name is still misleading as the intake is still contacting the cube after we suck it in, but this function will return `False`.

auscompgeek replied:
Well, at least the docstring actually describes what this is actually used for now.

What would be a more appropriate and concise name for this method?

ArthurAllshire replied:
Something like `wheels_contacting_cube`?

auscompgeek replied:
It returns a boolean, so it should be phrased in the form of a question. `are_wheels_contacting_cube` seems unwieldy.

ArthurAllshire replied:
What if we made some of these single line functions into properties?

auscompgeek replied:
Not really much of a fan of using a property here. Not sure how expensive the getOutputCurrent call is.

ArthurAllshire commented on components/intake.py:
```diff
@@ -31,61 +33,40 @@ def on_disable(self):
-        """Gets the distance of the infrared sensor in m."""
-        self.cube_distance = self.infrared.getDistance() / 100
-        return self.cube_distance
+    def get_cube_distance(self) -> float:
```
We should really note the failure condition for the infrared somewhere (you were saying something about it returning 35 the other day?) @Matthew13479

auscompgeek commented on automations/intake.py:
```diff
@@ -1,80 +1,72 @@
         """Deposit cube."""
-        if self.intake.seeing_cube():
+        if self.intake.is_seeing_cube():
             self.next_state("push_out_cube")
```
Now that I actually know what that `seeing_cube` method was intended to do, this state seems wrong.

auscompgeek committed:
intake: Improve naming, fix state machine

auscompgeek committed:
intake: Make arms_out tunable

This should make it easier to test extending the arms out from the
starting configuration, if need be.

auscompgeek committed:
intake: Use magicbot feedback for cube distance
